### PR TITLE
CS @ JHU 

### PR DESCRIPTION
**Pull request by danyaljj**
The CS primary faculty are on this page: https://www.cs.jhu.edu/faculty/
and the CS-affiliate faculty are here: https://www.cs.jhu.edu/people/cs-affiliates/

# Contributing to CSrankings

Thanks for contributing to CSrankings! Please read and indicate you agree with **all** these guidelines to getting your pull request accepted. Note that pull requests may take some time to get merged (please don't contact us for at least three months for updates, unless you are a sponsor - see below).

**If you find CSrankings useful, [please consider becoming a sponsor](https://github.com/sponsors/csrankings).**

**NOTE:** IF YOU DO NOT FOLLOW THE STEPS BELOW, YOUR COMMIT WILL BE SUMMARILY REJECTED.
You must read and check **all** the boxes below by filling them in with an X or your PR will be rejected.

**The Basics**

- [x] All pull requests and issues must come from non-anonymous accounts.

- [x] Use a reasonable title that explains what the PR corresponds to (as in, not "Update csrankings-x.csv").

- [x] Combine multiple updates to a single institution into a **single PR.**

- [x] Only submit one pull request per institution.

- [x] Do not modify any files except `csrankings-[a-z].csv` or (if needed) `country-info.csv` and `old/industry.csv` (see below).

- [x] Do not use Excel to edit any .csv files; Excel incorrectly tries to
convert some Google Scholar entries to formulas, corrupting the
database. Use the GitHub user interface or a text editor like emacs or NotePad instead.

- [x] Insert new faculty **in alphabetical order** (not at the end) in the appropriate `csrankings-[a-z].csv` files. **Do not modify `csrankings.csv`, which is auto-generated.**

- [x] Check to make sure that you have no spaces after commas, or any missing fields.

- [x] Check to make sure the home page is correct.

- [x] Make sure the Google Scholar IDs are just the alphanumeric identifier (not a URL or with `&hl=en`).

- [x] Check to make sure the name corresponds to the DBLP entry (look it up at http://dblp.org).

- [x] If a faculty member is not in a CS department or similar, include a comment explaining how they meet the inclusion criteria (see below).

**Inclusion criteria**

- [x] Make sure that any faculty you add meet the inclusion
criteria. Eligible faculty include only full-time, tenure-track research
faculty members on a given campus who can *solely* advise PhD students in
Computer Science. Faculty not in a CS department or similar who can
advise PhD students in CS can be included regardless of their home
department. **Provide justification, pointing to specific home pages
showing how faculty not in a CS department meet the inclusion criteria,
e.g. showing a courtesy appointment in CS.** Faculty must also have a 75%+ time appointment (check
`old/industry.csv` for faculty who are now more than 25% in industry).

**Updating an affiliation or home page**

- [x] Update affiliations, home pages, and Google Scholar entries by modifying `csrankings-[a-z].csv`. For the Google Scholar entry, just use the alphanumeric identifier in the middle of the URL. If none is there, put `NOSCHOLARPAGE`.

**Adding one or more faculty members (including an entire department)**

- [x] If the department is not yet listed in CSrankings, the entire CS faculty needs to be added (not just one faculty member).

- [x] Enter each faculty member's [DBLP](http://dblp.org) name, home page, and Google Scholar entry (just the alphanumeric identifier, not the whole URL) by modifying `csrankings-[a-z].csv` (**the letters correspond to the first letter of the faculty members' names**); include disambiguation suffixes like `0001` as needed. If the faculty entry is currently ambiguous, please do not include them. Send mail to the DBLP maintainers (dblp@dagstuhl.de) with a few publications by a particular faculty member; also, open an issue so that when the DBLP database is updated, that faculty member's information can be added.

- [x] If DBLP has multiple entries for this person, *all of them need to be listed*. Do not update `dblp-aliases.csv`.

- [x] If the institution you are adding is not in the US,
update `country-info.csv` and add *all* of the faculty in the CS department.

**(Advanced) Quick contribution via a shallow clone** 

We recommend that you use the GitHub web user interface to make changes. However, it may be more convenient
to clone the repository for larger-scale changes.

However, a full clone of the CSrankings repository is almost 2GB.  To
contribute a change without creating a full local clone of the
CSrankings repo, you can perform a _shallow clone_.  To do so, follow these
steps:

1. Fork the CSrankings repo.  If you have an existing fork, but it is not up to date with the main repository, this technique may not work.  If necessary, delete and re-create your fork to get it up to date.  (Do **not** delete your existing fork if it has unmerged changes you want to preserve!)

1. Do a shallow clone of your fork: `git clone --depth 1 https://github.com/yourusername/CSrankings`.  This will only download the most recent commit, not the full git history.

1. Make your changes on a branch, push them to your clone, and create a pull request on GitHub as usual.

If you want to make another contribution and some time has passed, perform these steps again, creating a fresh fork and shallow clone.

### CODE REVIEW

**Comment by emeryberger** [comment]
What is the status of this PR? (It still says "WIP") Thanks!

**Comment by danyaljj** [comment]
Still work in progress. Will finish the update by July/August and update the thread. Thank you for the reminder!

File: csrankings-c.csv [hunk 1]
```diff
@@ -585,7 +586,7 @@ Chiehyeon Lim,UNIST,http://service.unist.ac.kr,NW7F08MAAAAJ
 Chien-Chao Tseng,National Chiao Tung University,https://win.cs.nctu.edu.tw/cctseng,ownGePIAAAAJ
 Chien-Chung Shen,University of Delaware,https://www.eecis.udel.edu/~cshen,QyipVRoAAAAJ
 Chien-Ju Ho,Washington University in St. Louis,http://chienjuho.com,DWKoeW0AAAAJ
-Chien-Ming Huang,Johns Hopkins University,https://www.cs.jhu.edu/~cmhuang,0kTel90AAAAJ
+Chien-Ming Huang 0001,Johns Hopkins University,https://www.cs.jhu.edu/~cmhuang,0kTel90AAAAJ
```
Comment (danyaljj) [inline csrankings-c.csv:602]: Not sure why CI is throwing error here: 
```
 Invalid name (Chien-Ming Huang 0001). Please ensure it matches the DBLP entry.
```
I can see the name disambiguated with "0001": https://dblp.org/pid/56/3778-1.html

File: csrankings-m.csv [hunk 2]
```diff
@@ -1142,6 +1142,7 @@ Mathias Lécuyer,University of British Columbia,https://mathias.lecuyer.me,WeIvM
 Mathias Lux,University of Klagenfurt,http://www.itec.uni-klu.ac.at/~mlux,ucLHfl4AAAAJ
 Mathias Niepert,University of Stuttgart,http://www.matlog.net,p5vLzq0AAAAJ
 Mathias Payer,EPFL,https://nebelwelt.net,9cFCY5wAAAAJ
+Mathias Unberath,Johns Hopkins University,https://mathiasunberath.github.io,1bd5mdIAAAAJ
```
Comment (danyaljj) [inline csrankings-m.csv:1162]: FYI @mathiasunberath

File: csrankings-d.csv [hunk 3]
```diff
@@ -1469,7 +1469,7 @@ Duong Hieu Phan,Ecole Normale Superieure,http://www.di.ens.fr/users/phan,8n2aNu8
 Durga Toshniwal,IIT Roorkee,http://www.iitr.ac.in/departments/CSE/pages/People+Faculty+Durga_Toshniwal.html,NOSCHOLARPAGE
 Dusit Niyato,Nanyang Technological University,http://www.ntu.edu.sg/home/dniyato,T8sVhLMAAAAJ
 Dusit Tao Niyato,Nanyang Technological University,http://www.ntu.edu.sg/home/dniyato,T8sVhLMAAAAJ
-Dustin Richmond,Univ. of California - Santa Cruz,https://www.dustinrichmond.com/,U9hK8g4AAAAJ
+Dustin Richmond,Univ. of California - Santa Cruz,https://www.dustinrichmond.com/,U9hK8g4AAAAJ
```
Comment (danyaljj) [inline csrankings-d.csv]: Nothing changed here. Not sure why it shows up in the diff ...

Comment (emeryberger) [reply csrankings-d.csv]: Don't worry about it.

**Comment by emeryberger** [comment]
I'd like to clear this PR. Please make any needed changes for me to merge. Thanks.

**Comment by danyaljj** [comment]
Thanks @emeryberger!  Ignoring the test failure, the PR looks good to go from side. Will wait for your feedback or green light.

**Comment by danyaljj** [comment]
The CS primary faculty are on this page: https://www.cs.jhu.edu/faculty/ 
and the CS-affiliate faculty are here: https://www.cs.jhu.edu/people/cs-affiliates/

**Comment by danyaljj** [comment]
If there is anything I can improve here, please let me know. @emeryberger

File: csrankings-z.csv [hunk 4]
```diff
@@ -286,6 +286,7 @@ Zia Khan,University of Maryland - College Park,https://www.cs.umd.edu/~zia,gJkhF
 Ziad Al-Halah,University of Utah,https://users.cs.utah.edu/~ziad/,pEL6l3cAAAAJ
 Ziad Kobti,University of Windsor,http://www.uwindsor.ca/science/computerscience/1060/dr-ziad,-CUZPJYAAAAJ
 Ziawasch Abedjan,University of Hannover,https://www.pi.uni-hannover.de/de/dbs/team/prof-dr-ziawasch-abedjan,QUdlpN0AAAAJ
+Ziang Xiao,Johns Hopkins University,https://www.ziangxiao.com/,MjkODLEAAAAJ
```
Comment (danyaljj) [inline csrankings-z.csv:300]: FYI @ziangxiao

File: csrankings-a.csv [hunk 5]
```diff
@@ -1713,6 +1715,7 @@ Anita Wasilewska,Stony Brook University,http://www3.cs.stonybrook.edu/~anita,NOS
 Anitha Gollamudi,University of Massachusetts Lowell,https://scholar.harvard.edu/anithag,6cLC6vYAAAAJ
 Anja Feldmann [INF],Max Planck Society,https://www.mpi-inf.mpg.de/inet/people/anja-feldmann,mSK3340AAAAJ
 Anja Lehmann,Hasso Plattner Institute,https://hpi.de/das-hpi/personen/professoren/prof-dr-anja-lehmann.html,zH3fwYoAAAAJ
+Anjalie Field,Johns Hopkins University,https://anjalief.github.io/,WQO8DHQAAAAJ
```
Comment (danyaljj) [inline csrankings-a.csv:1732]: FYI: @anjalief

**Comment by danyaljj** [comment]
@emeryberger I have addressed the issue that was causing this PR to fail. Please give it a look.

**Comment by emeryberger** [comment]
Can you see why this is now failing?

**Comment by danyaljj** [comment]
Hmm I see several `Expecting value: line 2 column 1 (char 1)` errors, though not clear to me what the cause is. I will need to look deeper, but if you have any hints let me know.

```
Processing Rachel Karchin,Johns Hopkins University,https://karchinlab.org/,_3axEQ8AAAAJ
Processing failed (Expecting value: line 2 column 1 (char 1)).
Processing Ramalingam Chellappa,Johns Hopkins University,https://engineering.jhu.edu/ece/faculty/rama-chellappa,mZxsTCcAAAAJ
Processing failed (Expecting value: line 2 column 1 (char 1)).
Processing Rebecca Schulman,Johns Hopkins University,https://engineering.jhu.edu/faculty/rebecca-schulman/,segdJQYAAAAJ
Processing failed (Expecting value: line 2 column 1 (char 1)).
Processing Renjie Zhao,Johns Hopkins University,https://renjiezhao.github.io/,2IrtUS4AAAAJ
Processing failed (Expecting value: line 2 column 1 (char 1)).
Processing Steven Salzberg,Johns Hopkins University,https://engineering.jhu.edu/faculty/steven-salzberg/,sUVeH-4AAAAJ
Processing failed (Expecting value: line 2 column 1 (char 1)).
Processing Vishal M. Patel 0001,Johns Hopkins University,https://engineering.jhu.edu/faculty/vishal-patel,AkEXTbIAAAAJ
Processing failed (Expecting value: line 2 column 1 (char 1)).
Processing Ziang Xiao,Johns Hopkins University,https://www.ziangxiao.com/,MjkODLEAAAAJ
Processing failed (Expecting value: line 2 column 1 (char 1)).
```

File: validate_commit.py [hunk 6]
```diff
@@ -112,8 +113,13 @@ def matching_name_with_dblp(name: str) -> int:
     dblp_url = f'https://dblp.org/search/author/api?q=author%3A{author_name}$%3A&format=json&c=10'
     try:
         # Send a request to the DBLP API.
+        print("dblp_url: ", dblp_url)
         response = requests.get(dblp_url)
         # Extract the number of completions from the JSON response.
+        if "<title>429 Too Many Requests</title>" in response.text:
+            # wait for a few seconds and try again
+            time.sleep(10)
+            return matching_name_with_dblp(name)
```
Comment (danyaljj) [inline validate_commit.py:121]: Notice this small addition to the validation script: if DBLP returns a timeout message, we wait and repeat.

**Comment by danyaljj** [comment]
So I dug deeper and figure that the error is stemming from `validate_commit.py`. Specifically, `matching_name_with_dblp` was failing because we made too many requests to DBLP: I am adding many missing names and instead of returning a JSON, it is returning an HTML error: 
```
dblp_url:  https://dblp.org/search/author/api?q=author%3AZiang%20Xiao:$%3A&format=json&c=10
response:  <Response [429]>
response.text:  
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html>
<head>
<title>429 Too Many Requests</title>
</head>
<body>
<h1>Too Many Requests</h1>
<p>The vast number of requests from your side has
led to a temporary timeout for you. Please try
again after waiting some time.</p>
<p>You should always be fine by waiting for at least
a second between to consecutive calls to this server's
webpages or its APIs. When downloading the big data
dump files, you should request at most one file per
minute.</p>
<p><a href="/faq/1474706.html">For
more information on crawling dblp please consult our
F.A.Q..</a></p>
</body>
</html>
```

To resolve the issue, I added a condition to wait and retry DBLP if returns timeout message. 
https://github.com/emeryberger/CSrankings/blob/b3bdb15dcaa3ff220f2e56c3485de8fecb31e1c7/validate_commit.py#L118-L121

**Pull request merged**